File: invesdwin-context-python-parent/invesdwin-context-python-runtime-contract/src/test/java/de/invesdwin/context/r/runtime/contract/InputsAndResultsTestBoolean.R
print("getBoolean")
getBoolean <- putBoolean
print(typeof(getBoolean))
print(getBoolean)
if(typeof(getBoolean) != "logical"){
	stop("getBoolean not logical!")
}

print("getBooleanVector")
getBooleanVector <- putBooleanVector
print(typeof(getBooleanVector))
print(getBooleanVector)
if(typeof(getBooleanVector) != "logical"){
	stop("getBooleanVector not logical!")
}

print("getBooleanVectorAsList")
getBooleanVectorAsList <- putBooleanVectorAsList
print(typeof(getBooleanVectorAsList))
print(getBooleanVectorAsList)
if(typeof(getBooleanVectorAsList) != "logical"){
	stop("getBooleanVectorAsList not logical!")
}

print("getBooleanMatrix")
getBooleanMatrix <- putBooleanMatrix
print(typeof(getBooleanMatrix))
print(getBooleanMatrix)
if(typeof(getBooleanMatrix) != "logical"){
	stop("getBooleanMatrix not logical!")
}

print("getBooleanMatrixAsList")
getBooleanMatrixAsList <- putBooleanMatrixAsList
print(typeof(getBooleanMatrixAsList))
print(getBooleanMatrixAsList)
if(typeof(getBooleanMatrixAsList) != "logical"){
	stop("getBooleanMatrixAsList not logical!")
}

File: invesdwin-context-python-parent/invesdwin-context-python-runtime-contract/src/test/java/de/invesdwin/context/python/runtime/contract/InputsAndResultsTestBoolean.py
print("getBoolean")
if 'getBoolean' in locals():
	raise Exception("getBoolean already defined!")
getBoolean = putBoolean
getBooleanType = type(getBoolean)
print(getBooleanType)
print(getBoolean)
if(getBooleanType is not bool):
	raise Exception("getBoolean not bool!")

print("getBooleanVector")
if 'getBooleanVector' in locals():
	raise Exception("getBooleanVector already defined!")
getBooleanVector = putBooleanVector
getBooleanVectorType = type(getBooleanVector[0])
print(getBooleanVectorType)
print(getBooleanVector)
if(getBooleanVectorType is not bool):
	raise Exception("getBooleanVector not bool!")

print("getBooleanVectorAsList")
if 'getBooleanVectorAsList' in locals():
	raise Exception("getBooleanVectorAsList already defined!")
getBooleanVectorAsList = putBooleanVectorAsList
getBooleanVectorAsListType = type(getBooleanVectorAsList[0])
print(getBooleanVectorAsListType)
print(getBooleanVectorAsList)
if(getBooleanVectorAsListType is not bool):
	raise Exception("getBooleanVectorAsList not bool!")

print("getBooleanMatrix")
if 'getBooleanMatrix' in locals():
	raise Exception("getBooleanMatrix already defined!")
getBooleanMatrix = putBooleanMatrix
getBooleanMatrixType = type(getBooleanMatrix[0][0])
print(getBooleanMatrixType)
print(getBooleanMatrix)
if(getBooleanMatrixType is not bool):
	raise Exception("getBooleanMatrix not bool!")

print("getBooleanMatrixAsList")
if 'getBooleanMatrixAsList' in locals():
	raise Exception("getBooleanMatrixAsList already defined!")
getBooleanMatrixAsList = putBooleanMatrixAsList
getBooleanMatrixAsListType = type(getBooleanMatrixAsList[0][0])
print(getBooleanMatrixAsListType)
print(getBooleanMatrixAsList)
if(getBooleanMatrixAsListType is not bool):
	raise Exception("getBooleanMatrixAsList not bool!")
